lib/using_restricted_array.rb

```diff
@@ -1,82 +1,140 @@
-require_relative 'restricted_array.rb'
+require_relative "restricted_array.rb"
+
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity:linear or O(n), n is the value_to_find of items in the array or the length of the input array
```

Comment:
Why? Also, since you mention `value_to_find` I assume you're intending for this comment to be the time complexity analysis for all the linear methods?

Reply:
(For all the other complexity analyses, please also explain why.)